src/server/pfs/server/driver.go

```diff
@@ -421,6 +424,57 @@ func (d *driver) deleteRepo(txnCtx *txncontext.TransactionContext, repo *pfs.Rep
 	return nil
 }
 
+func (d *driver) createProject(ctx context.Context, req *pfs.CreateProjectRequest) error {
+	if err := ancestry.ValidateName(req.Project.Name); err != nil {
+		return errors.Wrapf(err, "invalid project name")
+	}
+	if err := d.env.TxnEnv.WithWriteContext(ctx, func(txnCtx *txncontext.TransactionContext) error {
+		projects := d.projects.ReadWrite(txnCtx.SqlTx)
+		if req.Update {
+			if err := projects.Get(pfsdb.ProjectKey(req.Project), &pfs.ProjectInfo{}); err != nil {
```

Comment:
Collections have an update function that will return a not exists error if it doesn't exist.